File: projeto_brademo/lib/model/filme.dart
class Filme {
  final int idFilme;
  final String titulo;
  final String descricao;
  final String imagem;
  final int duracao;
  final String streaming;
  final String diretor;
  final String fotoDiretor;
  final String atores;
  final double nota;
  final String nacionalidade;
  final String classificacao;
  final List<String> generos;

  Filme({
    required this.idFilme,
    required this.titulo,
    required this.descricao,
    required this.imagem,
    required this.duracao,
    required this.streaming,
    required this.diretor,
    required this.fotoDiretor,
    required this.atores,
    required this.nota,
    required this.nacionalidade,
    required this.classificacao,
    required this.generos,
  });

  factory Filme.fromJson(Map<String, dynamic> json) {
    return Filme(
      idFilme: json['id_filme'],
      titulo: json['titulo_filme'],
      descricao: json['descricao_filme'],
      imagem: json['linkImagemFilme_filme'],
      duracao: json['duracao_filme'],
      streaming: json['streaming_filme'],
      diretor: json['diretor_filme'],
      fotoDiretor: json['imagemDiretor_filme'],
      atores: json['atorEspecifico_filme'],
      nota: (json['nota_filme'] as num).toDouble(),
      nacionalidade: json['nacionalidade_filme'],
      classificacao: json['classificacao_filme'],
      generos: List<String>.from(json['generos'] ?? []),
    );
  }
}

File: moviematch_back/src/main/java/com/brademobackend/moviematch/model/Filme.java
package com.brademobackend.moviematch.model;

import java.util.List;

import jakarta.persistence.ElementCollection;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Filme {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id_filme;
    private String titulo_filme;
    private String descricao_filme;
    private String imagemDiretor_filme;
    private Integer duracao_filme;
    private String streaming_filme;
    private String nacionalidade_filme;
    private String classificacao_filme;
    private String diretor_filme;
    private String linkImagemFilme_filme;
    private String atorEspecifico_filme;
    private double nota_filme;
    

    @ElementCollection
    private List<String> generos;
}